addons/crm/models/crm_lead.py

```diff
@@ -240,6 +240,28 @@ def _compute_partner_id_values(self):
         for lead in self:
             lead.update(lead._preare_values_from_partner(lead.partner_id))
 
+    @api.depends('partner_id.email')
+    def _compute_email_from(self):
+        for lead in self:
+            if lead.partner_id and lead.partner_id.email != lead.email_from:
+                lead.email_from = lead.partner_id.email
```

Comment:
Also check that partner has an email (False should not rewrite an existing value)